src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -29,58 +29,110 @@ namespace Microsoft.Sarif.Viewer.ErrorList
     public class ErrorListService
     {
         public static readonly ErrorListService Instance = new ErrorListService();
+        private const string VersionRegexPattern = @"\""version\"":\s*\""(?<version>[\d.]+)\""";
 
         public static void ProcessLogFile(string filePath, Solution solution, string toolFormat = ToolFormat.None)
         {
             SarifLog log = null;
 
             string logText;
             string outputPath = null;
+            bool saveOutputFile = true;
 
             if (toolFormat.MatchesToolFormat(ToolFormat.None))
             {
                 logText = File.ReadAllText(filePath);
-                string pattern = @"""version""\s*:\s*""1.0.0""";
-                Match match = Regex.Match(logText, pattern, RegexOptions.Compiled | RegexOptions.Multiline);
+                Match match;
+                
+                // Get the schema version of the unmodified input log
+                using (StringReader reader = new StringReader(logText))
+                {
+                    // Read the first 200 characters.
+                    char[] buffer = new char[200];
+                    reader.ReadBlock(buffer, 0, buffer.Length);
+                    match = Regex.Match(new string(buffer), VersionRegexPattern, RegexOptions.Compiled);
+                }
 
-                if (match.Success)
+                if (match?.Success == true)
```

Comment:
>? [](start = 25, length = 1)

No need for the null conditional operator (and thus, no need for the explicit comparison against `true`. `Regex.Match` is guaranteed to return a non-null `Match` object.